apps/api/tests/test_server_accessible_projects.py

```
@@ -584,6 +584,38 @@ async def test_add_mcp_entity_requires_project_for_restricted_credentials() -> N
     add.assert_not_awaited()
 
 
+@pytest.mark.asyncio
+async def test_add_mcp_entity_clamps_conflict_threshold_floor_for_mcp_callers() -> None:
+    ctx = McpContext(org_id=str(uuid4()), user_id=str(uuid4()), scopes=["mcp"])
+    add = AsyncMock(return_value=AddResponse(success=True, id="task_123", message="ok"))
```

Ruff (0.15.12)

[error] Undefined name `AddResponse`

(F821)